### DISCUSSION TITLE
I am trying to use both item and optional file upload functions in FastAPI, but it is not working properly. My model is TextData, which includes a mode and a data list. I expected my request to receive TextData and optional file uploads, but I encountered an issue.

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#9484] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
router = APIRouter()


class TextData(BaseModel):

    mode: Optional[Literal["fast", "accurate", "default"]] = Field(...)

    data: List[Dict[str, Any]] = Field(...)

    class Config:
        schema_extra = {
            "example": {
                "data": [{
                    "text": "平原上的火焰宣布延期上映"
                }, {
                    "text": "第十四届全运会在西安举办"
                }]
            }
        }


@router.post("/")
async def create_upload_file(
        data: TextData,
        custom_dict: Optional[UploadFile] = File("None"),
):
    if not custom_dict:
        return {"mode": data.mode, "data": data, "custom_dict": custom_dict.filename}
    return {"mode": data.mode, "data": data}
```

### Description

I am trying to use both item and optional file upload functions in FastAPI, but it is not working properly. My model is TextData, which includes a mode and a data list. I expected my request to receive TextData and optional file uploads, but I encountered an issue. Here is my request code:

### Operating System

Linux

### Operating System Details

I am trying to use both item and optional file upload functions in FastAPI, but it is not working properly. My model is TextData, which includes a mode and a data list. I expected my request to receive TextData and optional file uploads, but I encountered an issue. Here is my request code:

`
class TextData(BaseModel):
    mode: Optional[Literal["fast", "accurate", "default"]] = Field(...)
    data: List[Dict[str, Any]] = Field(...)
`

`
@router.post("/")
async def create_upload_file(
        data: TextData ,
        custom_dict: Optional[UploadFile] = File("None"),
):
`

request data:

[
  {
    "text": "trying to use both item and optional file upload functions in FastAPI"
  },
  {
    "text": "I expected my request to receive TextData and optional file uploads, but I encountered an issue"
  }
]

### FastAPI Version

0.95.1

### Python Version

python3.9

**Comment by jgould22** [comment]
It is helpful if you can provide a minimal example of what you are experiencing so that people can help you more easily.

Something copy and pastable like the below.
```python

from typing import Annotated, Optional, Literal, List, Dict, Any

import uvicorn
from fastapi import FastAPI, Query, Depends, File, UploadFile
from pydantic import BaseModel, Field

app = FastAPI()


class TextData(BaseModel):

    mode: Optional[Literal["fast", "accurate", "default"]] = Field(...)

    data: List[Dict[str, Any]] = Field(...)

    class Config:
        schema_extra = {
            "example": {"data": [{"text": "平原上的火焰宣布延期上映"}, {"text": "第十四届全运会在西安举办"}]}
        }


@app.post("/")
async def create_upload_file(
    data: TextData,
    custom_dict: Optional[UploadFile] = File("None"),
):
    if not custom_dict:
        return {"mode": data.mode, "data": data, "custom_dict": custom_dict.filename}
    return {"mode": data.mode, "data": data}

def main():
    uvicorn.run(app, host="0.0.0.0", port=8001)

if __name__ == "__main__":
    main()
```

How are you making the request?

**Comment by YuriiMotov** [comment, marked as answer]
You can't accept files and JSON body at the same time. You can use Form parameters instead

> You can declare multiple Form parameters in a path operation, but you can't also declare Body fields that you expect to receive as JSON, as the request will have the body encoded using application/x-www-form-urlencoded instead of application/json.
> 
> This is not a limitation of FastAPI, it's part of the HTTP protocol.


Source: https://fastapi.tiangolo.com/tutorial/request-forms/#about-form-fields
